python/python/lancedb/table.py

```diff
@@ -1464,14 +1466,19 @@ def search(
                 else:
                     raise e
 
-        return LanceQueryBuilder.create(
+        query_builder = LanceQueryBuilder.create(
             self,
             query,
             query_type,
             vector_column_name=vector_column_name,
             ordering_field_name=ordering_field_name,
         )
 
+        if with_row_id:
```

Comment:
Is exposing `with_row_id` safe?

Reply:
I don't think there is a problem exposing `with_row_id`.  However, I believe the user can already do:

```
my_tbl.search(...).with_row_id()
```

Do we need to add `with_row_id` to `search`?

Reply:
Ahh I see.. yeah this is redundant then. will update